src/App.tsx

```diff
@@ -1,14 +1,91 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+function filterBySelect(
+  selectValue: string,
+  todos: Todo[],
+) {
+  switch (selectValue) {
+    case 'active':
+      return todos
+        .filter(({ completed }) => completed === false);
+
+    case 'completed':
+      return todos
+        .filter(({ completed }) => completed === true);
+
+    default:
+      return todos;
+  }
+}
+
+function getTodo(todos: Todo[], id: number) {
+  return todos.find(todo => todo.id === id);
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [activeTodo, setActiveTodo] = useState<Todo | undefined>(undefined);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [isOpenModal, setIsOpenModal] = useState(false);
+
+  const handleQueryUpdate = (value: string) => {
+    setQuery(value);
+  };
+
+  const handleSelectUpdate = (value: string) => {
+    setSelectFilter(value);
+  };
+
+  const handleModal = (id?: number) => {
+    setIsOpenModal(current => !current);
+
+    if (id) {
+      const newActiveTodo = getTodo(todos, id);
+
+      setActiveTodo(newActiveTodo);
+      const fetchUser = async () => {
+        setActiveUser(await getUser(newActiveTodo?.userId));
+      };
+
+      fetchUser();
```

Comment:
why just do not make handleModal  as async?
```suggestion
   setActiveUser(await getUser(newActiveTodo?.userId));

```

and try to do smth like that, it would be clearer
```suggestion
   const user = await getUser(newActiveTodo?.userId);
    
   setActiveUser(user);

```